pytorch_lightning/trainer/trainer.py

```diff
@@ -449,17 +449,52 @@ def default_attributes(cls):
         return args
 
     @classmethod
-    def add_argparse_args(cls, parent_parser: ArgumentParser) -> ArgumentParser:
-        """Extend existing argparse by default `Trainer` attributes."""
-        parser = ArgumentParser(parents=[parent_parser], add_help=False)
+    def _get_argparse_args_and_types(cls) -> Generator[Tuple[str, Any, Any], None, None]:
```

Comment:
```suggestion
    def _get_init_arguments_and_types(cls) -> Generator[Tuple[str, Any, Any], None, None]:
```
sounds a bit better as it can be used elsewhere not only for `argparser`

Reply:
Maybe in this case it’s worth to make this method public?

Reply:
And also move the `allowed_types` to the method's input arguments?

Reply:
yes, both are good suggestions, lest doing it...

Reply:
I decided to make this method more general. Let's a client decide what arguments he needs. For instance, `add_argparse_args` is a "client" of this method, and it restricts the allowed types by itself. 
What are your thoughts on this?

Reply:
that would be nice, just thinking if is better to defien wanted args/types or ignored ones